mintpy/prep_hyp3.py

```diff
@@ -0,0 +1,221 @@
+    Inputs:
+        *unw_phase.tif or *corr.tif file name or *dem.tif file name, e.g.
+            S1AA_20161223T070700_20170116T070658_VVP024_INT80_G_ueF_74C2_unw_phase.tif
+            S1AA_20161223T070700_20170116T070658_VVP024_INT80_G_ueF_74C2_corr.tif
+            dem.tif
+    Output:
+        Metadata dictionary (meta)
+    '''
+
+    meta = {}
+
+    # Geospatial Data
+    ds = gdal.Open(fname, gdal.GA_ReadOnly)
+    meta['FILE_LENGTH'] = ds.RasterYSize
+    meta['LENGTH'] = ds.RasterYSize
+    meta['WIDTH'] = ds.RasterXSize
+    geoTrans = ds.GetGeoTransform()
+    del ds
+
+    x0 = geoTrans[0]
+    y0 = geoTrans[3]
+    x_step = geoTrans[1]
+    y_step = geoTrans[5]
+    x1 = x0 + x_step * meta['WIDTH']
+    y1 = y0 + y_step * meta['LENGTH']
+
+    meta['X_FIRST'] = f'{x0:.9f}'
+    meta['Y_FIRST'] = f'{y0:.9f}'
+    meta['X_STEP'] = f'{x_step:.9f}'
+    meta['Y_STEP'] = f'{y_step:.9f}'
+    meta['X_UNIT'] = 'METERS'
+    meta['Y_UNIT'] = 'METERS'
+
+    meta["LON_REF1"] = x0
+    meta["LON_REF2"] = x1
+    meta["LON_REF3"] = x0
+    meta["LON_REF4"] = x1
+
+    meta["LAT_REF1"] = y0
+    meta["LAT_REF2"] = y0
+    meta["LAT_REF3"] = y1
+    meta["LAT_REF4"] = y1
+
+    # Satellite platform data
+    #   Note: HyP3 currently only supports Sentinel-1 data, so Sentinel-1
+    #         configuration is hard-coded.
+    meta['PROCESSOR'] = 'hyp3'
+    meta['PLATFORM'] = 'Sen'
+    meta['ANTENNA_SIDE'] = -1
+    meta['WAVELENGTH'] = SPEED_OF_LIGHT / sensor.SEN['carrier_frequency']
+    meta['HEIGHT'] = 693000.0 # nominal altitude of Sentinel1 orbit
+
+    # Earth radius probably won't be used anywhere for the geocoded data
+    meta['EARTH_RADIUS'] = 6337286.638938101
+
+    return(meta)
+
+
+# Extract data from HyP3 interferogram metadata
+def add_ifg_metadata(fname,meta):
+    '''Read/extract attribute data from HyP3 metadata file and add to metadata dictionary
+    Inputs:
+        *unw_phase.tif or *corr.tif file name, e.g.
+            S1AA_20161223T070700_20170116T070658_VVP024_INT80_G_ueF_74C2_unw_phase.tif
+            S1AA_20161223T070700_20170116T070658_VVP024_INT80_G_ueF_74C2_corr.tif
+        Metadata dictionary (meta)
+    Output:
+        Metadata dictionary (meta)
+    '''
+    directory = os.path.dirname(fname)
+
+    sat, date1_string, date2_string, pol, res, soft, proc, ids, *_ = os.path.basename(fname).split('_')
+
+    job_id = '_'.join([sat, date1_string, date2_string, pol, res, soft, proc, ids])
+    date1 = datetime.strptime(date1_string,'%Y%m%dT%H%M%S')
+    date2 = datetime.strptime(date2_string,'%Y%m%dT%H%M%S')
+    date_avg = date1 + (date2 - date1) / 2
+
+    ifg_meta_file = f'{os.path.join(directory,job_id)}.txt'
+    ifg_meta = {}
+    with open(ifg_meta_file, 'r') as f:
+        for line in f:
+            key, value = line.strip().split(': ')
+            ifg_meta[key] = value
+
+    # Add relevant data to meta object
+    meta['CENTER_LINE_UTC'] = date_avg.strftime('%y%m%d')
```

Comment:
This attribute contains the time of the day only, in the unit of seconds, without date info. Here (https://github.com/insarlab/MintPy/blob/main/mintpy/utils/isce_utils.py#L159) is an example.

Reply:
float